core/src/main/scala/org/apache/spark/internal/Logging.scala

```diff
@@ -192,7 +211,15 @@ private[spark] object Logging {
         defaultSparkLog4jConfig = false
         LogManager.resetConfiguration()
       } else {
-        LogManager.getRootLogger().setLevel(defaultRootLevel)
+        val rootLogger = LogManager.getRootLogger()
+        rootLogger.setLevel(defaultRootLevel)
+        rootLogger.getAllAppenders().asScala.foreach { tmp =>
```

Comment:
Any reason not to just iterate through `consoleAppenderToThreshold.keys()`? Not a huge deal but cuts down on a bit of work

Reply:
Not really, consoleAppenderToThreshold was added later. Additionally, currently I do not store any consoleAppender which have null threshold into this map. In order for doing this way, I will need to store those as well. Dropping this, as in most scenarios, this will require more storage and minor performance improvement.